### ISSUE TITLE
NF Fails evaluating sum of iteration variables with -nonfScalarize

### ISSUE THREAD

**Issue by mscuttari** [OpenModelica/OpenModelica#9248]
### Description
The OpenModelica frontend crashes when evaluating parameters.

### Steps to reproduce
Given the attached .mo file, the frontend can't instantiate the model when run with

```
 omc PowerGridDAE.mo +i=PowerGridDAE.GridBase -f -d=nonfScalarize,arrayConnect,combineSubscripts,printRecordTypes,evaluateAllParameters --newBackend --showStructuralAnnotations
```

And emits the following error:

```
Error processing file: PowerGridDAE.mo
[/var/lib/jenkins2/ws/LINUX_BUILDS/tmp.build/openmodelica-1.20.0~dev-218-g2dd6f78/OMCompiler/Compiler/NFFrontEnd/NFCeval.mo:1034:9-1034:67:writable] Error: Internal error NFCeval.evalBinaryAdd failed to evaluate ‘i + j‘

# Error encountered! Exiting...
# Please check the error message and the flags.
```

Commenting lines from 76 to 82 solves the problem. However, without those lines the model is obviously incomplete.
It seems that the frontend can't evaluate the sum `i+j`, but the same doesn't hold for lines 49-50, which are processed correctly.

### Expected behavior
Model is instantiated correctly.

### Additional files
[PowerGridDAE.zip](https://github.com/OpenModelica/OpenModelica/files/9142528/PowerGridDAE.zip)

### Version and OS
 - OpenModelica 1.20.0~dev-218-g2dd6f78
 - Ubuntu 18.04

**Comment by casella** [comment]
A further hint, the NF can handle equations like
```modelica
v_start[i,j] = if mod(i+j,1) == 0 then Complex(Vg) else Vl;
```
that have such an expression in a _conditional expression_, while it fails when the expression shows up in _conditional equations_, such as
```modelica
if mod(i+j,2) == 1 then
  v[i,j]*CM.conj(i_n[i,j]) = Complex(P);
end if;
```

**Comment by perost** [comment]
Fixed with #9931.
